File: python/meteroid/models/metric_filter.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .metric_filter_operator import MetricFilterOperator


@dataclasses.dataclass
class MetricFilter(BaseModel):
    """A pre-aggregation filter: only events whose `property` matches feed the metric's
    aggregation. Distinct from a segmentation dimension (which splits pricing). Multiple
    filters are ANDed."""

    op: MetricFilterOperator

    property: str

    values: t.List[str]

File: go/metric_filter.go
// this file is @generated
package meteroid

// A pre-aggregation filter: only events whose `property` matches feed the metric's
// aggregation. Distinct from a segmentation dimension (which splits pricing). Multiple
// filters are ANDed.
type MetricFilter struct {
	Op MetricFilterOperator `json:"op"`

	Property string `json:"property"`

	Values RequiredSlice[string] `json:"values"`
}
